packages/flame_test/lib/src/flame_test.dart

```diff
@@ -17,15 +17,15 @@ extension FlameGameExtension on Component {
   /// Makes sure that the [component] is added to the tree if you wait for the
   /// returned future to resolve.
   Future<void> ensureAdd(Component component) async {
-    await add(component);
-    updateTree(0);
+    add(component);
+    await Component.root!.ready();
```

Comment:
Shouldn't it be enough to just await this `add` now if a future is returned from `add`?

Reply:
Not really: the future from `add` can only take you as far as to the end of the `onLoad` stage. But if you need to make sure that the component is actually mounted into the tree, `add()` cannot provide that.